### PR TITLE
Remove usr-merge compatibilty symlinked modules

### PR DESCRIPTION
**Pull request by silkeh**
Remove the corresponding symlink/directory in `/lib64/modules/` when modules are set to be installed in `/usr/`. This enables the migration path that Solus follows for the usr merge.

### CODE REVIEW

File: src/bootman/kernel.c [hunk 1]
```diff
@@ -27,6 +27,30 @@
 
 #include "config.h"
 
+static inline const char *strip_slashes(const char *path) {
```
Comment (EbonJaeger) [inline src/bootman/kernel.c:30]: This only strips the first instance of a slash, so the name is a bit misleading.

Comment (silkeh) [reply src/bootman/kernel.c]: Good catch! Renamed it to `strip_leading_slashes` (because paths can have multiple leading slashes for some reason).

**Comment by EbonJaeger** [comment]
What would the procedure be to test this? Also, are there any prerequisites, given some people are usr-merged, while some aren't?

**Comment by silkeh** [comment]
> What would the procedure be to test this?

Remove the latest kernel: 

```
sudo env CBM_DEBUG=1 ./build/src/clr-boot-manager remove-kernel com.solus-project.current.6.9.8-294
```

> Also, are there any prerequisites, given some people are usr-merged, while some aren't?

Nope. It should work the same on any form of merged and non-merged system.

**Comment by EbonJaeger** [comment]
Here's the output. Not sure if that's expected or not.

```bash
❯ sudo env CBM_DEBUG=1 ./build/src/clr-boot-manager remove-kernel com.solus-project.current.6.9.8-294
[DEBUG] cbm (../src/cli/cli.c:L142): No such file: //etc/kernel/update_efi_vars
[INFO] cbm (../src/bootman/bootman.c:L788): Current running kernel: 6.9.8-294.current
[INFO] cbm (../src/bootman/sysconfig.c:L179): Discovered UEFI ESP: /dev/disk/by-partuuid/839dacbb-0600-4289-a81f-1c7073484788
[INFO] cbm (../src/bootman/sysconfig.c:L256): Fully resolved boot device: /dev/nvme0n1p1
[DEBUG] cbm (../src/bootman/bootman.c:L141): shim-systemd caps: 0x26, wanted: 0x26
[DEBUG] cbm (../src/bootman/bootman.c:L156): UEFI boot now selected (shim-systemd)
remove-kernel takes a kernel ID of the form org.clearlinux.TYPE.VERSION-RELEASE
```

**Comment by EbonJaeger** [comment]
```
❯ sudo env CBM_DEBUG=1 ./build/src/clr-boot-manager remove-kernel com.solus-project.current.6.9.8-294
[DEBUG] cbm (../src/cli/cli.c:L142): No such file: //etc/kernel/update_efi_vars
[INFO] cbm (../src/bootman/bootman.c:L788): Current running kernel: 6.9.8-294.current
[INFO] cbm (../src/bootman/sysconfig.c:L179): Discovered UEFI ESP: /dev/disk/by-partuuid/839dacbb-0600-4289-a81f-1c7073484788
[INFO] cbm (../src/bootman/sysconfig.c:L256): Fully resolved boot device: /dev/nvme0n1p1
[DEBUG] cbm (../src/bootman/bootman.c:L141): systemd caps: 0x26, wanted: 0x26
[DEBUG] cbm (../src/bootman/bootman.c:L156): UEFI boot now selected (systemd)
[DEBUG] cbm (../src/bootman/kernel.c:L156): Falling back to pre-usr-merge module directory: /lib64/modules/6.8.12-293.current
[INFO] cbm (../src/bootman/bootman.c:L503): Checking for mounted boot dir
[INFO] cbm (../src/bootman/bootman.c:L556): Mounting boot device /dev/nvme0n1p1 at /boot
[SUCCESS] cbm (../src/bootman/bootman.c:L569): /dev/nvme0n1p1 successfully mounted at /boot
[DEBUG] cbm (../src/bootman/kernel.c:L767): Removing module dir: ///usr/lib64/modules/6.9.8-294.current
[DEBUG] cbm (../src/bootman/kernel.c:L780): Removing pre-usr-merged module dir: /lib64/modules/6.9.8-294.current
[INFO] cbm (../src/bootman/bootman.c:L469): Attempting umount of /boot
[SUCCESS] cbm (../src/bootman/bootman.c:L473): Unmounted boot directory
```

**Comment by silkeh** [comment]
@EbonJaeger: that looks correct, specifically:

```
[DEBUG] cbm (../src/bootman/kernel.c:L767): Removing module dir: ///usr/lib64/modules/6.9.8-294.current
[DEBUG] cbm (../src/bootman/kernel.c:L780): Removing pre-usr-merged module dir: /lib64/modules/6.9.8-294.current
```

Don't forget to reinstall your kernel!

**Review by EbonJaeger — APPROVED** [review]
This appears to work for me!

**Pull request merged**